ams/server/src/main/java/com/netease/arctic/server/optimizing/SchedulingPolicy.java

```diff
@@ -23,16 +23,22 @@ public class SchedulingPolicy {
   private static final String BALANCED = "balanced";
 
   private final Map<ServerTableIdentifier, TableRuntime> tableRuntimeMap = new HashMap<>();
-  private final Comparator<TableRuntime> tableSorter;
+  private Comparator<TableRuntime> tableSorter;
   private final Lock tableLock = new ReentrantLock();
 
   public SchedulingPolicy(ResourceGroup group) {
-    String schedulingPolicy = Optional.ofNullable(group.getProperties())
-            .orElseGet(Maps::newHashMap)
-            .get(SCHEDULING_POLICY_PROPERTY_NAME);
-    if (StringUtils.isBlank(schedulingPolicy) || schedulingPolicy.equalsIgnoreCase(QUOTA)) {
+    setTableSorterIfNeeded(group);
+  }
+
+  public void setTableSorterIfNeeded(ResourceGroup optimizerGroup) {
+    String schedulingPolicy = Optional.ofNullable(optimizerGroup.getProperties())
+        .orElseGet(Maps::newHashMap)
+        .get(SCHEDULING_POLICY_PROPERTY_NAME);
+    if ((StringUtils.isBlank(schedulingPolicy) || schedulingPolicy.equalsIgnoreCase(QUOTA)) &&
+        tableSorter == null || (tableSorter instanceof BalancedSorter)) {
```

Comment:
what's the purpose of the condition: `tableSorter instanceof BalancedSorter`?

Reply:
I assume it covers the situation that `tableSorter` is already initiated and is the same as the new configuration set.